sycl/include/sycl/ext/oneapi/experimental/invoke_simd.hpp

```diff
@@ -209,8 +211,15 @@ using SpmdRetType =
 
 template <class SimdCallable, class... SpmdArgs>
 static constexpr int get_sg_size() {
-  if constexpr (has_uniform_signature<SimdCallable, SpmdArgs...>()) {
-    return 0; // subgroup size does not matter then
+  if constexpr (all_uniform_types<SpmdArgs...>()) {
```

Comment:
I think this is a a good idea. But we should update the extension document to reflect this. It seems at the moment it is missing a description when the sg-size can be deduced. BTW now that this is implemented should the extension be moved to experimental?

Reply:
ok, I can create a PR, you can do this as well if you wish

> BTW now that this is implemented should the extension be moved to experimental?

This is in sycl::ext::oneapi::experimental already, am I missing something?

Reply:
> ok, I can create a PR, you can do this as well if you wish

happy for you to do it

I meant the location of https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_invoke_simd.asciidoc. I think this should be moved from proposed to experimental.

Reply:
> happy for you to do it

OK, I'll do this

>  I think this should be moved from proposed to experimental.

Oh, that's what you mean. Agree.